Userland/Applications/HexEditor/HexEditor.cpp

```diff
@@ -623,3 +623,37 @@ Vector<Match> HexEditor::find_all(ByteBuffer& needle, int start)
 
     return matches;
 }
+
+Vector<Match> HexEditor::find_all_strings(size_t min_length)
+{
+    if (m_buffer.is_empty())
+        return {};
+
+    Vector<Match> matches;
+
+    int offset = -1;
+    StringBuilder builder;
+    for (size_t i = 0; i < m_buffer.size(); i++) {
+        char c = m_buffer.bytes().at(i);
+        if (isprint(c)) {
+            if (offset == -1)
+                offset = i;
+            builder.append(c);
```

Comment:
Could we get away with just storing a `StringView` into `m_buffer` instead of allocating new strings for each match?

Reply:
I presume this would be unsafe as `m_buffer` can change without changing the `SearchResultsModel` and associated `TableView`. However you're probably right that a `StringView` can be used instead.

Reply:
but if the underlying buffer changes, the search results are invalid anyway :shrug:

Reply:
`SearchResultsModel` was added (yesterday) to facilitate 'Find All' and now 'Find All Strings' functionality in this PR.

It seems a better approach would be to store a start offset and end offset which point into `m_buffer`, rather than start offset and value.

`GUI::ModelRole::Display` in the model can take care of bounds checking and formatting for the `TableView`.

> but if the underlying buffer changes, the search results are invalid anyway shrug

My concern was coercing invalid bytes or out of bounds bytes into a string if the buffer had changed. But this could also be taken care of by using the `GUI::ModelRole::Display` logic in the model.

Reply:
But that should be done in a separate PR.

Reply:
Alright, sure.